src/rovio_rosbag_loader.cpp

```diff
@@ -0,0 +1,128 @@
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+*/
+
+#include <ros/package.h>
+#include <rosbag/bag.h>
+#include <rosbag/view.h>
+#include <memory>
+#include "rovio/RovioFilter.hpp"
+#include "rovio/RovioNode.hpp"
+#include <boost/foreach.hpp>
+#define foreach BOOST_FOREACH
+
+#ifdef ROVIO_NMAXFEATURE
+static constexpr int nMax_ = ROVIO_NMAXFEATURE;
+#else
+static constexpr int nMax_ = 25; // Maximal number of considered features in the filter state.
+#endif
+
+#ifdef ROVIO_NLEVELS
+static constexpr int nLevels_ = ROVIO_NLEVELS;
+#else
+static constexpr int nLevels_ = 4; // // Total number of pyramid levels considered.
+#endif
+
+#ifdef ROVIO_PATCHSIZE
+static constexpr int patchSize_ = ROVIO_PATCHSIZE;
+#else
+static constexpr int patchSize_ = 8; // Edge length of the patches (in pixel). Must be a multiple of 2!
+#endif
+
+#ifdef ROVIO_NCAM
+static constexpr int nCam_ = ROVIO_NCAM;
+#else
+static constexpr int nCam_ = 1; // Used total number of cameras.
+#endif
+
+#ifdef ROVIO_NPOSE
+static constexpr int nPose_ = ROVIO_NPOSE;
+#else
+static constexpr int nPose_ = 0; // Additional pose states.
+#endif
+
+typedef rovio::RovioFilter<rovio::FilterState<nMax_,nLevels_,patchSize_,nCam_,nPose_>> mtFilter;
+
+int main(int argc, char** argv){
+  ros::init(argc, argv, "rovio");
+  ros::NodeHandle nh;
+  ros::NodeHandle nh_private("~");
+
+  std::string rootdir = ros::package::getPath("rovio"); // Leaks memory
+  std::string filter_config = rootdir + "/cfg/rovio.info";
+
+  nh_private.param("filter_config", filter_config, filter_config);
+
+  // Filter
+  std::shared_ptr<mtFilter> mpFilter(new mtFilter);
+  mpFilter->readFromInfo(filter_config);
+
+  // Force the camera calibration paths to the ones from ROS parameters.
+  for (unsigned int camID = 0; camID < nCam_; ++camID) {
+    std::string camera_config;
+    if (nh_private.getParam("camera" + std::to_string(camID)
+                            + "_config", camera_config)) {
+      mpFilter->cameraCalibrationFile_[camID] = camera_config;
+    }
+  }
+  mpFilter->refreshProperties();
+
+  // Node
+  rovio::RovioNode<mtFilter> rovioNode(nh, nh_private, mpFilter);
+  rovioNode.makeTest();
+
+  rosbag::Bag bag;
+  bag.open("/home/michael/datasets/euroc/01_easy/dataset.bag", rosbag::bagmode::Read);
+
+  std::vector<std::string> topics;
+  topics.push_back(std::string("/imu0"));
```

Comment:
Avoid using leading slashes: then you can use this on real helicopter datasets since if you run this node in a namespace it will resolve correctly (i.e., "bluebird/imu0" instead of "/imu0") without having to do anything differently.